src/libraries/System.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/MarshalTests.cs

```diff
@@ -0,0 +1,571 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices.JavaScript;
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Runtime.InteropServices.JavaScript.Tests
+{
+    public static class MarshalTests
+    {
+        [Fact]
+        public static void MarshalPrimitivesToCS()
+        {
+            HelperMarshal._i32Value = 0;
+            Runtime.InvokeJS("App.call_test_method(\"InvokeI32\", [10, 20])");
+            Assert.Equal(30, HelperMarshal._i32Value);
+
+            HelperMarshal._f32Value = 0;
+            Runtime.InvokeJS("App.call_test_method(\"InvokeFloat\", [1.5])");
+            Assert.Equal(1.5f, HelperMarshal._f32Value);
+
+            HelperMarshal._f64Value = 0;
+            Runtime.InvokeJS("App.call_test_method(\"InvokeDouble\", [4.5])");
+            Assert.Equal(4.5, HelperMarshal._f64Value);
+
+            HelperMarshal._i64Value = 0;
+            Runtime.InvokeJS("App.call_test_method(\"InvokeLong\", [99])");
+            Assert.Equal(99, HelperMarshal._i64Value);
+        }
+
+        [Fact]
+        public static void MarshalArrayBuffer()
+        {
+            Runtime.InvokeJS(@"
+                var buffer = new ArrayBuffer(16);
+                App.call_test_method (""MarshalArrayBuffer"", [ buffer ]);
```

Comment:
nit: in the previous test you don't have a space after a method name

Reply:
Addressed in commits